editor/src/core/shared/github/helpers.ts

```diff
@@ -912,97 +973,104 @@ export function getRefreshGithubActions(
+      if (branchOriginChecksums == null) {
+        if (originCommitSha != null) {
+          // 3. get the checksums
           promises.push(
-            updatePullRequestsForBranch(operationContext)(
-              dispatch,
-              githubRepo,
-              branchName,
-              initiator,
-            ),
+            getBranchChecksums(operationContext)(githubRepo, branchName, originCommitSha),
           )
         }
+      } else {
+        // 4. update PRs
+        promises.push(
+          updatePullRequestsForBranch(operationContext)(
+            dispatch,
+            githubRepo,
+            branchName,
+            initiator,
+          ),
+        )
       }
-    } else {
-      promises.push(Promise.resolve([updateGithubData({ branches: null })]))
     }
+    // 5. finally update upstream changes
+    promises.push(
+      updateUpstreamChanges(
+        branchName,
+        branchOriginChecksums,
+        githubRepo,
+        previousCommitSha,
+        operationContext,
+      ),
+    )
+  } else {
+    promises.push(Promise.resolve([updateGithubData({ branches: null })]))
   }
 
   return promises
 }
 
 export async function refreshGithubData(
   dispatch: EditorDispatch,
-  githubAuthenticated: boolean,
   githubRepo: GithubRepo | null,
   branchName: string | null,
   branchOriginChecksums: FileChecksumsWithFile | null,
-  githubUserDetails: GithubUser | null,
   previousCommitSha: string | null,
   originCommitSha: string | null,
   operationContext: GithubOperationContext,
   initiator: GithubOperationSource,
 ): Promise<void> {
-  const promises = await getRefreshGithubActions(
+  const promises = getRefreshGithubActions(
     dispatch,
-    githubAuthenticated,
     githubRepo,
     branchName,
     branchOriginChecksums,
-    githubUserDetails,
     previousCommitSha,
     originCommitSha,
     operationContext,
     initiator,
   )
-  // Resolve all the promises.
-  await Promise.allSettled(promises).then((results) => {
-    let actions: Array<EditorAction> = []
-    for (const result of results) {
-      switch (result.status) {
-        case 'rejected':
-          console.error(`Error while polling Github: ${result.reason}`)
-          break
-        case 'fulfilled':
-          actions.push(...result.value)
-          break
-      }
+
+  // Resolve all the promises (sequentially).
+  const results = await sequentialPromiseAllSettled(promises)
+
+  let actions: Array<EditorAction> = []
+  for (const result of results) {
+    switch (result.status) {
+      case 'rejected':
+        console.error(`Error while polling Github: ${result.reason}`)
+        break
+      case 'fulfilled':
+        actions.push(...result.value)
+        break
     }
+  }
 
-    // Dispatch all the actions from all the polling functions.
-    dispatch(actions, 'everyone')
-  })
+  // Dispatch all the actions from all the polling functions.
+  dispatch(actions, 'everyone')
+}
+
+async function sequentialPromiseAllSettled<T>(
```

Comment:
:eyes:

Reply:
updated the code to _fire_ the promises sequentially in https://github.com/concrete-utopia/utopia/pull/5389/commits/3e669a8eb6017f7b928fb39bbd494e45212c2d81